compiler/damlc/daml-doc/src/DA/Daml/Doc/Driver.hs

```diff
@@ -18,67 +21,99 @@ import Development.IDE.Types.Location
 import Development.IDE.Types.Diagnostics
 import Development.IDE.Types.Options
 
-import           Control.Monad.Extra
-import           Control.Monad.Except
+import Control.Monad.Extra
+import Control.Monad.Except
+import Data.Maybe
+import System.IO
+import System.Exit
+import System.Directory
+import System.FilePath
+
 import qualified Data.Aeson                        as AE
 import qualified Data.Aeson.Encode.Pretty          as AP
-import qualified Data.ByteString.Lazy              as BS
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as LBS
 import qualified Data.Text                         as T
 import qualified Data.Text.Encoding                as T
-import           System.Directory
-import           System.FilePath
 
+data DamldocOptions = DamldocOptions
+    { do_inputFormat :: InputFormat
+    , do_ideOptions :: IdeOptions
+    , do_outputPath :: FilePath
+    , do_outputFormat :: OutputFormat
+    , do_docTemplate :: Maybe FilePath
+    , do_docOptions :: [DocOption]
```

Comment:
Having a type called `DamldocOptions` that has a list of `DocOption` seems confusing. We should probably refactor the whole handling of `DocOption` at some point. But that’s probably better left for a separate PR.

Reply:
I agree :-)